.github/workflows/tests.yml

```diff
@@ -98,3 +98,14 @@ jobs:
         shell: bash
         env:
           GH_TOKEN: ${{ github.token }}
+
+  spelling:
+   name: Spell Check with Typos
+   runs-on: ubuntu-latest
+   steps:
+   - name: Checkout Actions Repository
+     uses: actions/checkout@v4
+   - name: Spell Check Repo
+     uses: crate-ci/typos@v1.29.10
+     with:
+       files: ./AUTHORS ./CMakeLists.txt ./docs/ ./README.md ./tests/ ./toolchains
```

Comment:
I think this would be better in this file: https://github.com/gyselax/gyselalibxx/blob/main/.github/workflows/static_analyses.yml

with the static analysis tests that are not costly and are therefore always run (even when the PR is in draft)